api/src/pcapi/core/offerers/models.py

```diff
@@ -899,6 +899,38 @@ def first_user(self) -> "users_models.User | None":
                 return user_offerer.user
         return None
 
+    @property
+    def hasValidBankAccount(self) -> bool:
+        return any(
+            (
+                bank_account.status is finance_models.BankAccountApplicationStatus.ACCEPTED
+                for bank_account in self.bankAccounts
+            )
+        )
+
+    @property
+    def hasPendingBankAccount(self) -> bool:
+        return any(
+            (
+                bank_account.status is finance_models.BankAccountApplicationStatus.ON_GOING
+                for bank_account in self.bankAccounts
+            )
+        )
+
+    @property
+    def venuesWithNonFreeOffersWithoutBankAccounts(self) -> list[int]:
+        """
+        Don’t use this property without proper eager loading or you could
+        face huge N+1 problem (see `offerer/{id}/bank-accounts`)
+        """
+        ids: list[int] = []
+        for venue in self.managedVenues:
+            if any((link.timespan.upper is None for link in venue.bankAccountLinks)):
```

Comment:
**Remarque :** Ici la condition `upper is None` reste valide par rapport à ce que permet aujourd'hui l'interface, mais ne le sera pas si un jour on peut avoir `[dans un mois, None)` ou encore `[hier, dans un an)` en base, avec une période de validité des CB. Est-ce ok ?

Reply:
Comme un collègue m’avait très judicieusement conseillé de fetch les `VenueBankAccountLink` selon la condition `timespan.contains(utcnow())` on prévient ce genre de cas ici :) @prouzet